internal/engine/compiler/RATIONALE.md

```diff
@@ -58,6 +58,18 @@ And `nativecall` is actually implemented in [arch_amd64.s](./arch_amd64.s)
 as a convenience layer to comply with the Go's official calling convention.
 We delegate the task to jump into the code segment to the Go assembler code.
 
+
+## Why it's safe to execute runtime-generated machine codes against Goroutine preemption
+
```

Comment:
there is a little jargon, so possibly an overview sentence or two can help.

Maybe start with the problem of goroutine premption, and then say "unsafe" means goroutines are not allowed to pre-empt.